src/Microsoft.AspNet.Mvc.TagHelpers/JavaScriptHelpers.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+	internal static class JavaScriptHelpers
+    {
+        private static readonly Assembly ResourcesAssembly = typeof(JavaScriptHelpers).GetTypeInfo().Assembly;
+        private static readonly ConcurrentDictionary<Tuple<string, bool>, string> Cache = new ConcurrentDictionary<Tuple<string, bool>, string>();
+        
+        /// <summary>
+        /// Gets an embedded JavaScript file resource and optionally decodes it for use as a .NET format string.
+        /// </summary>
+    	public static string GetEmbeddedJavaScript(string resourceName, bool isFormatString = false)
```

Comment:
Unless I'm missing a call site it looks as if we're only ever calling this with `isFormatString: true`. Might be too preemptive to even provide an option to override that functionality. I'd rather add the capability to provide the "if is format string" functionality in an appropriate PR where it's used. Therefore for now we can just assume `true`, should simplify the `Cache` keys and this method.